charts/pega/charts/hazelcast/templates/_helpers.tpl

```diff
@@ -0,0 +1,36 @@
+{{- define "hazelcastName" -}} hazelcast {{- end -}}
+{{- define "hazelcastEnvironmentConfig" -}} hz-env-config {{- end -}}
+{{- define "hazelcastConfig" -}} hz-config {{- end -}}
+
+
+{{- define "isHazelcastEnabled" }}
+ {{- if .Values.enabled -}}
+  true
+ {{- else -}}
+  false
+ {{- end -}}
+{{- end }}
+
+{{- define "waitForHazelcast" -}}
+- name: wait-for-hazelcast
+  image: busybox:1.31.0
+  # Init container for waiting for hazelcast service to initialize.
+  args:
+  - sh
+  - -c
+  - >
+    set -e
+    counter=0;
+    while [ -z $(wget -S "http://{{ template "hazelcastName" . }}-service.{{ .Release.Namespace }}:5701/hazelcast/health/cluster-safe" 2>&1 | grep "HTTP/" | awk '{print $2}') ] || [ $(wget -q -O - "http://{{ template "hazelcastName" . }}-service.{{ .Release.Namespace }}:5701/hazelcast/health/cluster-size" /dev/null) -ne {{ .Values.hazelcast.replicas }} ] || [ $(wget -S "http://{{ template "hazelcastName" . }}-service.{{ .Release.Namespace }}:5701/hazelcast/health/cluster-safe" 2>&1 | grep "HTTP/" | awk '{print $2}') -ne 200 ]; do
+    echo "waiting for hazelcast pods to start and join the cluster..." ;
+    counter=$(($counter+5));
+    sleep 5;
+    if [ $counter -gt 150 ]; then
+    echo "Timeout Reached. Hazelcast pods failed to join the cluster";
```

Comment:
It might be worth clarifying this message.  To pass this test, we just need at least one hz server node up and running, right?  Maybe change this to: "Timeout Reached. Hazelcast pods are not available"

Also doesn't cluster-safe check to see if all data migrations are complete.  Is this required before allowing clients to connect to the hz cluster?

Reply:
Right it is 
/hazelcast/health/Cluster-State(status should be 200) and not the /hazelcast/health/cluster-safe

Reply:
ok...I changed it to cluster-state.

Reply:
It is a recommendation from hazelcast to have minimum 3 servers for a cluster, so we want HZ cluster to be formed before client connects to the cluster.

Reply:
Is it possible to avoid hard-coding endpoints and just take the full URL? An issue with Stream Service just came up last week where we had originally [hardcoded the endpoint](https://github.com/pegasystems/docker-pega-web-ready/blob/1bc9c990317294f35e3da06f08fb1945bf8b255e/tomcat-conf/Catalina/localhost/prweb.xml#L20) (taking in the base URL as a param), but they needed to change that. This now means we have to make changes to the helm charts/docker image. If we had used the full URL as the param we wouldn't have had to make any changes.

Reply:
So in this case, we need to construct the URL dynamically using service name and namespace. Namespace would only be available at run time. Also this is at the helm level. So are you suggesting to have this endpoint defined in some local variable and then refer that local variable in if condition , instead of doing a check on partially hardcoded url.
We tested this helm charts using infinity SCE and didn't see any issues.

Also in the docker image changes , we are passing complete URL from helm config maps and we don't need to append any additional path params to it.  See this PR:-
https://github.com/pegasystems/docker-pega-web-ready/pull/83

Reply:
I think ideally we don't have to know about the details of these services configuration. Getting the namespace at deployment time makes sense, but if we could avoid the specifics of their endpoint in our code somehow I think that would be ideal. I'm not suggesting there's any problem with the way things work now, just trying to think of ways to avoid tight coupling between services. I think providing a default endpoint and allowing it to be overridden like you suggest sounds good.